sdk/core/Azure.Core.Amqp/src/AmqpDataBody.cs

```diff
@@ -11,21 +11,21 @@ namespace Azure.Core.Amqp
     /// This consists of one or more data sections.
     /// <seealso href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-data" />
     /// </summary>
-    public class AmqpDataMessageBody : AmqpMessageBody
+    public class AmqpDataBody : AmqpMessageBody
     {
         /// <summary>
-        /// Initializes a new <see cref="AmqpDataMessageBody"/> instance with the
+        /// Initializes a new <see cref="AmqpDataBody"/> instance with the
         /// passed in data sections.
         /// </summary>
         /// <param name="data">The data sections.</param>
-        public AmqpDataMessageBody(IEnumerable<BinaryData> data)
+        public AmqpDataBody(IEnumerable<ReadOnlyMemory<byte>> data)
         {
             Data = data;
         }
 
         /// <summary>
         /// The data sections for the AMQP message body.
         /// </summary>
-        public IEnumerable<BinaryData> Data { get; internal set; }
+        public IEnumerable<ReadOnlyMemory<byte>> Data { get; internal set; }
```

Comment:
I'm a bit confused why we are stopping using BinaryData. Is it just because it is a class now?